test_util/integration_test.py

```diff
@@ -1260,11 +1260,23 @@ def wait_for_endpoint():
 
     wait_for_endpoint()
 
-    test_cache_url = "http://{}:{}/signal_test_cache".format(service_points[0].host, service_points[0].port)
+    test_host = service_points[0].host
+    test_port = service_points[0].port
+    test_cache_url = "http://{}:{}/signal_test_cache".format(test_host, test_port)
+
     cmd = """
-data=`/opt/mesosphere/bin/dcos-signal -report-host leader.mesos -test 2> /dev/null`;
-curl -H 'Content-Type: application/json' -X POST -d "$data" {};
+cleanup() {
```

Comment:
This cleanup will never get hit. Note we're also running inside a mesos task sandbox here. I think echo `"test_id" > test-cluster-id && /opt/mesosphere/bin/dcos-sign -cluster-id-path $PATH/test-cluster-id` (and the rest) would work better.